core/src/types/token.rs

```diff
@@ -320,12 +318,17 @@ impl From<Denomination> for u8 {
 )]
 pub struct DenominatedAmount {
     /// The mantissa
-    pub amount: Amount,
+    amount: Amount,
     /// The number of decimal places in base ten.
-    pub denom: Denomination,
+    denom: Denomination,
 }
 
 impl DenominatedAmount {
+    /// Make a new denominated amount
```

Comment:
I would rework the docstr here to make it a bit more clear that what we're returning is `amount * 10^denom`